tidb-cloud/vector-search-index.md

```diff
@@ -0,0 +1,244 @@
+---
+title: Vector Search Index
+summary: Learn how to build and use the vector search index to accelerate K-Nearest neighborhood (KNN) queries in TiDB.
+---
+
+# Vector Search Index
+
+K-Nearest neighborhood (KNN) search is the problem of finding the K closest points for a given point in a vector space. The most straightforward approach to solve this problem is a brute force search where the distance between all points in the vector space and the reference point is computed. This method guarantees perfect accuracy, but it is usually too slow for practical applications. Thus, nearest neighborhood search problems are often solved with approximative algorithms.
+
+In TiDB, you can create and utilize Vector Search Indexes for such approximate nearest neighbor (ANN) searches over columns with [Vector Data Types](/tidb-cloud/vector-search-data-types.md). By using Vector Search Indexes, vector search queries could be finished in milliseconds.
+
+TiDB currently supports the following Vector Search Index algorithms:
+
+- HNSW
+
+> **Note:**
+>
+> Vector Search index is only available for [TiDB Serverless](/tidb-cloud/select-cluster-tier.md#tidb-serverless) clusters.
+
+## Create HNSW Vector Index
+
+HNSW ([wikipedia](https://en.wikipedia.org/wiki/Hierarchical_navigable_small_world)) is one of the most popular vector indexing algorithms. HNSW index provides a good performance with a relatively high accuracy (> 98% in typical cases).
+
+To create a HNSW vector index, specify the index definition in the comment of a column with [vector data type](/tidb-cloud/vector-search-data-types.md) when creating the table:
+
+```sql
+CREATE TABLE vector_table_with_index (
+    id INT PRIMARY KEY, doc TEXT,
+    embedding VECTOR(3) COMMENT "hnsw(distance=cosine)"
+);
+```
+
+> **Note:**
+>
+> The syntax to create Vector Index may be changed in future releases.
+
+You must specify the distance metric via `distance=<metric>` config when creating the vector index:
+
+- Cosine Distance: `COMMENT "hnsw(distance=cosine)"`
+- L2 Distance: `COMMENT "hnsw(distance=l2)"`
+
+Vector index can be only created for fixed-dimensional vector columns like `VECTOR(3)`. It cannot be created for mixed-dimensional vector columns like `VECTOR`, because vector distances can be only calculated between vectors with the same dimensions.
```

Comment:
```suggestion
The vector index can only be created for fixed-dimensional vector columns like `VECTOR(3)`. It cannot be created for mixed-dimensional vector columns like `VECTOR` because vector distances can only be calculated between vectors with the same dimensions.
```